homeassistant/components/freedompro/fan.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Freedompro fan."""
+import json
+
+from pyfreedompro import put_state
+
+from homeassistant.components.fan import FanEntity
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro fan."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "fan"
+    )
+
+
+class Device(CoordinatorEntity, FanEntity):
```

Comment:
Conside giving these classes more descriptive names (a general comment for all Freedom pro components):

```suggestion
class FreedomproFan(CoordinatorEntity, FanEntity):
```